### PR TITLE
Add Decorator classes

### PR DESCRIPTION
**Pull request by ANTOINE1128**
# In this PR I have completed the following requirements :
## Interface
- Create a class Nameable.

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @ANTOINE1128,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Some of the Table of Content links are broken. Kindly fix them or remove them.

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Tony, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Let's ensure that the ReadMe for this project is professionally documented by updating broken links, as indicated in the previous review:

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -19,3 +26,45 @@ def of_age?
...
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (mckent05) [inline person.rb]: - Good job done so far, but your correct_name method should return `@nameable.correct_name`. Please fix this 🙏

File: person.rb [hunk 2]
```diff
@@ -19,3 +26,45 @@ def of_age?
...
+class CapitalizeDecorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (mckent05) [inline person.rb]: - Good job done so far, but according to the project requirements,  please make sure your `CapitalizeDecorator` inherits from the `BaseDecorator`

File: person.rb [hunk 3]
```diff
@@ -19,3 +26,45 @@ def of_age?
...
+
+class TrimmerDecorator < Nameable
+  def initialize(nameable, max_length = 10)
+    @nameable = nameable
+    @max_length = max_length
+    super()
+  end
```
Comment (mckent05) [inline person.rb]: - Good job done so far, but according to the project requirements, please make sure your `TrimmerDecorator` inherits from the `BaseDecorator`. Please fix this 🙏

File: person.rb [hunk 4]
```diff
@@ -19,3 +26,45 @@ def of_age?
...
+
+  def correct_name
+    @nameable.correct_name.slice(0, @max_length)
+  end
+end
```
Comment (mckent05) [inline person.rb]: **OPTIONAL** - Good job done so far, but in order to make your code cleaner and easier to read, I suggest you put all Decorators in their individual files. It'd help make your code more modular, please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding decorators to your application 👏 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 5]
```diff
@@ -1,9 +1,50 @@
...
+require_relative 'Nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name.capitalize
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [x] just as the previous reviewer mentioned it should be ` @nameable.correct_name` not ` @nameable.correct_name.capitalize`. kindly fix that

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @ANTOINE1128 , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 5 above)
Comment (Abenezer-Tilahun) [reply person.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @ANTOINE1128 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**